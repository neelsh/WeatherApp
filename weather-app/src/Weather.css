body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

input[type="text"] {
  padding: 10px;
  border: solid 1px #fff;
  box-shadow: inset 1px 1px 2px 0 #707070;
  transition: box-shadow 0.3s;
}

.focus {
  margin-left: 10px;
}

.focus :focus {
box-shadow: inset 1px 1px 2px 0 #c9c9c9;
}

.wrapper {
  margin-top: 17px;
  margin-bottom: 10px;
}

.wrapper, .current {
  text-align: center;
}

.weather {
  font-size: 30px;
  text-decoration: underline;
}

.temp-wrapper, .cond-wrapper, .suggest-wrapper {
  font-size: 22px;
  padding-bottom: 7px;
}

.suggest-wrapper {
  margin-bottom: 70px;
}

.temp, .condition, .suggestion {
  padding-left: 9px;
}

.forecast {
  border-top: 4px solid pink;
  padding-top: 50px;
}
