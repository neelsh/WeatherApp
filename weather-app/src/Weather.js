import React from 'react';
import './Weather.css';
import xhr from 'xhr';
import Plot from './Plot';

class Weather extends React.Component {
  state = {
    location: '',
    data: {},
    dates: [],
    temps: [],
  };

  fetchData = (evt) => {
    evt.preventDefault();
    console.log('fetch data for', this.state.location);
    const location = encodeURIComponent(this.state.location);
    const urlPrefix = 'http://api.openweathermap.org/data/2.5/forecast?q=London,GB';
    const urlSuffix = '&APPID=ed85c941348b145ed73ab007fd218d19&units=metric';
    const url = urlPrefix + location + urlSuffix;
    const self = this;

    xhr({
      url: url
    }, function (err, data) {
        const body = JSON.parse(data.body);
        const list = body.list;
        const dates = [];
        const temps = [];
        for (var i = 0; i < list.length; i++) {
          dates.push(list[i].dt_txt);
          temps.push(list[i].main.temp);
        }

        self.setState({
          data: body,
          dates: dates,
          temps: temps,
        })
    });
  };

  changeLocation = (evt) => {
    this.setState({
      location: evt.target.value
    })
  }

  render() {
    let currentTemp = 'not loaded yet';
    if(this.state.data.list) {
      currentTemp = this.state.data.list[0].main.temp;
    }
    let currentCondition = 'condition not loaded yet';
    if(this.state.data.list) {
      currentCondition = this.state.data.list[0].weather[0].main;
    }
    let currentSuggestion = 'No suggestion yet...';
    if(currentCondition.indexOf('Rain')>=0) {
      currentSuggestion = 'Pack an umbrella...'
    } else {
      currentSuggestion = 'No need to pack an umbrella today!'
    }
    return (
      <div>
        <div className="current">
          <h1 className="weather">Weather</h1>
          <form onSubmit={this.fetchData}>
            <h3>I want to know the weather for
              <button className="focus" placeholder=":focus"
                value={this.state.location}
                onChange={this.changeLocation}>London</button>
            </h3>
          </form>
        </div>
        {(this.state.data.list) ? (
          <div className="wrapper">
            <p className="temp-wrapper">
            Current Temperature:
            <span className="temp">{ currentTemp }</span>
            <span className="temp-symbol">Â°C</span>
            </p>
            <p className="cond-wrapper">
            Current Condition:
            <span className="condition">{ currentCondition }</span>
            </p>
            <p className="suggest-wrapper">
            Suggestion:
            <span className="suggestion">{ currentSuggestion }</span>
            </p>
            <h2 className="forecast">Forecast for the next 4 days</h2>
              <Plot
                xData={this.state.dates}
                yData={this.state.temps}
                type="scatter"
              />
          </div>
        ) : null }
      </div>
    )
  }
}

export default Weather;
