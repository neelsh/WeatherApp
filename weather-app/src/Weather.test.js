import React from 'react';
import {shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Weather from './Weather';
import currentTemp from './Weather';

describe('weather app', () => {
  it('should render the weather app', () => {
    const component = renderer.create(<Weather />).toJSON;
    expect(component).toMatchSnapshot();
  });

  it('should have a location of London', () => {
    const weather = shallow(<Weather />)
    weather.find('button').simulate('click');
    expect(weather.find('.focus').text()).toEqual('London')
  });

  it('displays the temperature', () => {
    const weather = shallow(<Weather />)
    weather.find('temp')
    expect(weather.find('currentTemp')).toEqual(temp)
  });

  it('displays the condition', () => {
    const weather = shallow(<Weather />)
    weather.find('condition')
    expect(weather.find('currentCondition')).toEqual(condition)
  });
})
