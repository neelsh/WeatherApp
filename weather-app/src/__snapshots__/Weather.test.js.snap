// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`weather app should render the weather app 1`] = `[Function]`;
